### PR TITLE
#308: Rename KOKKOS macros

### LINKED ISSUE #308: Rename macros prefixed with `KOKKOS_`
All macros defined and used in this library should be prefixed with this library's name. Kokkos is encountering a lot of trouble with Trilinos packages that inappropriately define `KOKKOS_FOO` macros for their own use. Don't be like them.

### PR DESCRIPTION
**Pull request by thearusable**
Closes #308

### CODE REVIEW

**Review by JacobDomagala — APPROVED** [review]

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -96,7 +96,7 @@ if (${kokkos_DIR_FOUND})
   endif()
   target_compile_definitions(
     ${CHECKPOINT_LIBRARY} PUBLIC
-    KOKKOS_ENABLED_CHECKPOINT=1 KOKKOS_KERNELS_ENABLED=${KERNELS}
+    MAGISTRATE_ENABLED_KOKKOS=1 MAGISTRATE_ENABLED_KOKKOS_KERNELS=${KERNELS}
```
Comment (cz4rs) [inline src/CMakeLists.txt]: We usually go with `_ENABLED` at the end in `vt` (for example `vt_libfort_enabled`). Consider this for consistency:
```suggestion
    MAGISTRATE_KOKKOS_ENABLED=1 MAGISTRATE_KOKKOS_KERNELS_ENABLED=${KERNELS}
```

**Review by cz4rs — APPROVED** [review]
Looks good to me.

File: src/CMakeLists.txt (hunk 1 above)
Comment (thearusable) [reply src/CMakeLists.txt]: Fixed.

**Review by cz4rs — APPROVED** [review]

**Comment by PhilMiller** [comment]
Make sure that application code wasn't relying on any of these macros, or that it gets updated with any shift to a newer version of Magistrate

**Review by lifflander — APPROVED** [review]
I don't think that apps are depending on this macro.

**Pull request merged**